src/main/java/io/jenkins/plugins/conventionalcommits/NextVersionStep.java

```diff
@@ -100,7 +102,8 @@ protected String run() throws Exception {
                     }
                 }
 
-                Version currentVersion = Version.valueOf(latestTag.isEmpty() ? "0.0.0" : latestTag);
+                Version currentVersion = new GetCurrentVersion().getCurrentVersion(dir, latestTag);
```

Comment:
Love how this is very generic and the major logic is kept out of the main class!